### PR TITLE
Update Add Contact Drawer Panel

### PR DESCRIPTION
**Pull request by Full-lifey**
Updates the Add Contact Panel to match the wireframes. Also builds out the verify discard changes panel and the confirm add contact panel.

### CODE REVIEW

File: src/Types/generateOptions.ts [hunk 1]
```diff
@@ -17,7 +17,7 @@ export type DefaulOptions = {
   priorities: string[];
   doy: number;
   pass: number;
-  passes?: { id: string; aos: string; los: string }[];
+  passes: { id: string; aos: string; los: string }[];
```
Comment (Full-lifey) [inline src/Types/generateOptions.ts:20]: I'm not sure why this was optional as it's built when first ran. It was causing a lot of downstream checks for existence.

File: src/components/App.css [hunk 2]
```diff
@@ -1,6 +1,6 @@
 .App-main {
   --right-panel-width: 26.25rem;
-  --top-deduction: 8rem;
+  --top-deduction: 6rem;
```
Comment (Full-lifey) [inline src/components/App.css:3]: This was causing a large gap on my screen at the bottom. I tested this by resizing and this size puts the container + padding at the bottom correctly.

File: src/components/App.css [hunk 3]
```diff
@@ -11,7 +11,7 @@
 }
 
 .App-main__container {
-  height: 100%;
+  height: calc(100% - 44px);
```
Comment (Full-lifey) [inline src/components/App.css:14]: The container was overflowing the parent causing some bottom content to get cut off like a horizonal scroll bar, the last item in the list, and the footers in the right panel.

File: src/components/GlobalStatusBar/GlobalStatusBar.tsx [hunk 4]
```diff
@@ -10,9 +10,24 @@ import {
...
+};
+
+const range: RangeItem[] = [
+  { threshold: 17, status: 'off' },
+  { threshold: 33, status: 'critical' },
+  { threshold: 50, status: 'serious' },
+  { threshold: 66, status: 'caution' },
+  { threshold: 83, status: 'standby' },
+  { threshold: 100, status: 'normal' },
+];
```
Comment (Full-lifey) [inline src/components/GlobalStatusBar/GlobalStatusBar.tsx:29]: I added this `RangeItem` array to solve errors in the console.

File: src/components/ManageContacts/ContactDetails.tsx [hunk 5]
```diff
@@ -2,26 +2,22 @@ import {
...
 
 type PropTypes = {
-  label?: string;
-  value: string;
+  handleAction: (action?: Actions) => void;
 };
 
-const ReadOnlyInput = ({ label, value }: PropTypes) => (
-  <RuxInput label={label} readonly value={value} size='small' />
-);
```
Comment (Full-lifey) [inline src/components/ManageContacts/ContactDetails.tsx:22]: Moved this input up to a common component as it is used at least 25 times. It only saves two props from the overall code but it seemed worth it with such large usage.

**Review by Paterma — CHANGES_REQUESTED** [review]
Great work. A few things though- it throws this error when you click "Add Contact" on the confirm screen and the page goes blank, so right now adding a contact doesn't work.
Screenshot 2023-06-08 at 3 09 23 PM

There's a few style things as well, like the alignment of the icons. Are we going to do the arrows, too?
Screenshot 2023-06-08 at 3 14 11 PM


The confirm screen needs some adjustment on the text and buttons. 
Screenshot 2023-06-08 at 3 15 01 PM


What about the pass range section?
Screenshot 2023-06-08 at 3 19 21 PM

**Comment by Full-lifey** [comment]
> Great work. A few things though- it throws this error when you click "Add Contact" on the confirm screen and the page goes blank, so right now adding a contact doesn't work. Screenshot 2023-06-08 at 3 09 23 PM
> 
> There's a few style things as well, like the alignment of the icons. Are we going to do the arrows, or details, too? Screenshot 2023-06-08 at 3 14 11 PM
> 
> The confirm screen needs some adjustment on the text and buttons. Screenshot 2023-06-08 at 3 15 01 PM
> 
> What about the pass range section? Screenshot 2023-06-08 at 3 19 21 PM

1. Error resolved. Needed to force the user away from the confirm panel
2. The arrows and select icon to render inputs functionality was too much to add to this PR. There are too many pieces here to include in this already large body of work.
3. Confirm screen has been adjusted
4. The pass range section is functionality that doesn't make sense for the depth of our data as well as being inputs that don't exist in astro. I made the decision not to include those inputs.

**Review by Paterma — APPROVED** [review]

**Review by NicDominguez — APPROVED** [review]

**Pull request merged**